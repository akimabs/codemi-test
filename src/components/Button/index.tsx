import React from 'react';
import {
  Animated,
  TouchableNativeFeedback,
  TouchableOpacity,
  View,
  Platform,
} from 'react-native';
import {Text} from '..';

import styles from './styles';

type PropsButton = {
  title?: String;
  type?: 'primary' | 'danger' | 'light' | 'success';
  onPress?: () => void;
  style?: Object;
  children?: any;
  withOutRipple?: boolean;
  withOutAnimate?: boolean;
  disable?: boolean;
};

const Button: React.FC<PropsButton> = ({
  type,
  style,
  title,
  onPress,
  children,
  withOutRipple = false,
  withOutAnimate = false,
  disable = false,
}) => {
  const [state] = React.useState({
    animated: new Animated.Value(1),
  });

  const inAnimate = (): void => {
    Animated.spring(state.animated, {
      toValue: 1.03,
      useNativeDriver: true,
    }).start();
  };

  const outAnimate = (): void => {
    Animated.spring(state.animated, {
      toValue: 1,
      useNativeDriver: true,
    }).start(onPress);
  };

  const animatedStyle: Object = {
    transform: [{scale: state.animated}],
  };

  const types: any = [
    type === 'danger' && styles.dark,
    type === 'primary' && styles.primary,
    type === 'light' && styles.light,
    type === 'success' && styles.success,
  ];

  const textStyles: any = [
    type === 'danger' && styles.textDark,
    type === 'primary' && styles.textPrimary,
    type === 'light' && styles.textLight,
    type === 'success' && styles.textDark,
  ];

  const Touchable: any =
    Platform.OS !== 'ios'
      ? withOutRipple
        ? TouchableOpacity
        : TouchableNativeFeedback
      : TouchableOpacity;

  return (
    <View>
      {disable ? (
        <View>
          {withOutAnimate ? (
            <View style={[types, style]}>
              {children ? (
                children
              ) : (
                <Text type="bold" style={textStyles}>
                  {title}
                </Text>
              )}
            </View>
          ) : (
            <Animated.View style={[animatedStyle, types, style]}>
              {children ? (
                children
              ) : (
                <Text type="bold" style={textStyles}>
                  {title}
                </Text>
              )}
            </Animated.View>
          )}
        </View>
      ) : (
        <Touchable
          activeOpacity={0.99}
          delayPressIn={0}
          useForeground={true}
          onPressIn={inAnimate}
          onPressOut={outAnimate}
          onPress={onPress}>
          {withOutAnimate ? (
            <Animated.View style={[types, style]}>
              {children ? (
                children
              ) : (
                <Text type="bold" style={textStyles}>
                  {title}
                </Text>
              )}
            </Animated.View>
          ) : (
            <Animated.View
              // eslint-disable-next-line react-native/no-inline-styles
              style={[{overflow: 'hidden'}, animatedStyle, types, style]}>
              {children ? (
                children
              ) : (
                <Text type="bold" style={textStyles}>
                  {title}
                </Text>
              )}
            </Animated.View>
          )}
        </Touchable>
      )}
    </View>
  );
};

export default Button;
