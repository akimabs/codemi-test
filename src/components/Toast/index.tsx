import Toast from 'react-native-toast-message';

let toast: any;

export default function showToast(
  message: string,
  type: any,
  duration = 4000,
  offset = 100,
) {
  Toast.show({
    type,
    position: 'top',
    text1: 'Attention',
    text2: message,
    visibilityTime: duration,
    autoHide: true,
    topOffset: offset,
  });
}

export function hideLoading() {
  Toast.hide(toast);
}

export function showErrorToast(message: string, offset?: number) {
  showToast(message, 'error', offset);
}

export function showSuccessToast(message: string, offset?: number) {
  showToast(message, 'success', offset);
}

export function showInfoToast(message: string, offset?: number) {
  showToast(message, 'info', offset);
}
