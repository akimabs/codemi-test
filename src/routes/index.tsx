import 'react-native-gesture-handler';

import React from 'react';
import {createStackNavigator, TransitionPresets} from '@react-navigation/stack';
import {NavigationContainer} from '@react-navigation/native';
import {connect} from 'react-redux';

// Splash screen
import SplashScreen from '../screens/auth/splash-screen';

// Onboarding screen
import StartedScreen from '../screens/auth/started-screen';

// Login screen
import Login from '../screens/auth/login';

// Stack screen
import Home from '../screens/home';
import Scanner from '../screens/qr-scan/';
import DataScan from '../screens/qr-scan/data-scan';

function Private({auth, apps}: any) {
  const SplashStack: any = createStackNavigator();
  const OnboardingStack: any = createStackNavigator();
  const AuthStack: any = createStackNavigator();
  const StackNav: any = createStackNavigator();

  if (apps.isFirst) {
    return (
      <NavigationContainer>
        <SplashStack.Navigator
          initialRouteName="SplashScreen"
          headerMode="none">
          <SplashStack.Screen name="SplashScreen" component={SplashScreen} />
        </SplashStack.Navigator>
      </NavigationContainer>
    );
  }

  if (!auth.isDoneOnboarding) {
    return (
      <NavigationContainer>
        <OnboardingStack.Navigator
          screenOptions={TransitionPresets.ModalSlideFromBottomIOS}
          initialRouteName="SplashScreen"
          headerMode="none">
          <OnboardingStack.Screen
            name="StartedScreen"
            options={{
              ...TransitionPresets.ModalSlideFromBottomIOS,
            }}
            component={StartedScreen}
          />
        </OnboardingStack.Navigator>
      </NavigationContainer>
    );
  }

  if (!auth.token) {
    return (
      <NavigationContainer>
        <AuthStack.Navigator
          screenOptions={TransitionPresets.ScaleFromCenterAndroid}
          initialRouteName="SplashScreen"
          headerMode="none">
          <AuthStack.Screen name="Login" component={Login} />
        </AuthStack.Navigator>
      </NavigationContainer>
    );
  }

  return (
    <NavigationContainer>
      <StackNav.Navigator
        screenOptions={TransitionPresets.SlideFromRightIOS}
        initialRouteName="Home"
        headerMode="none">
        <StackNav.Screen name="Home" component={Home} />
        <StackNav.Screen name="Scanner" component={Scanner} />
        <StackNav.Screen
          options={{
            ...TransitionPresets.ModalSlideFromBottomIOS,
          }}
          name="DataScan"
          component={DataScan}
        />
      </StackNav.Navigator>
    </NavigationContainer>
  );
}

const mapStateToProps = (state: any) => ({
  auth: state.auth,
  apps: state.apps,
});

export default connect(mapStateToProps)(Private);
