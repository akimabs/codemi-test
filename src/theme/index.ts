const colors = {
  primary: '#32ab3a',
  primary2: '#2a9631',
  white: '#FFF',
  black1: '#000',
  black2: '#4B4F71',
  gray1: '#9C9C9C',
  gray2: '#c4c4c4',
  gray3: '#e6eaed',
  gray4: '#677176',
  gray5: '#8f8f8f',
  backgroundColor: '#F4F4F4',
  info: '#62B1F6',
  success: '#5cb85c',
  danger: '#d9534f',
  warning: '#f0ad4e',
  warning2: '#f96d01',
  dark: '#000',
  dark2: '#393939',
  light: '#f4f4f4',
  statusBar: '#F2C94C',
  like: '#59abe3',
};

const images = {
  Onboarding: {
    image1: require('../../assets/images/onboarding1.png'),
    image2: require('../../assets/images/onboarding2.png'),
    image3: require('../../assets/images/onboarding3.png'),
  },
  Logo: require('../../assets/images/Logo.png'),
};

const sizes = {
  radiusSmall: 5,
  radiusMedium: 8,
  radiusHigh: 10,
};

export {colors, sizes, images};
