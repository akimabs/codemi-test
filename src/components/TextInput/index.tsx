import React from 'react';
import {TextInput, Animated, View, Platform, Dimensions} from 'react-native';
import {colors, sizes} from '../../theme';
import Icon from 'react-native-vector-icons/Feather';
import {Button} from '..';
import {heightPercentageToDP} from 'react-native-responsive-screen';
import {RFValue as fs} from 'react-native-responsive-fontsize';

const {width} = Dimensions.get('window');

type PropsText = {
  placeholder?: string;
  style?: Object;
  containerStyle?: Object;
  handle: Function;
  isPassword?: boolean;
  isNumber?: any;
  multiline?: boolean;
  uppercase?: boolean;
};

const CustomTextInput: React.FC<PropsText> = ({
  style,
  containerStyle,
  placeholder,
  handle,
  isPassword,
  isNumber,
  multiline,
  uppercase,
}) => {
  const [state, setState] = React.useState({
    value: '',
    animated: new Animated.Value(0),
    hide: true,
  });

  const handleType = async (value: string) => {
    await setState({
      ...state,
      value,
    });
    await handle(value);
  };

  React.useEffect(() => {
    state.value !== '' ? animateText(fs(7, width)) : animateText(0);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state.value.length === 0 ? state.value : null]);

  const animateText = (toValue: number): void => {
    Animated.timing(state.animated, {
      toValue,
      duration: 300,
      useNativeDriver: false,
    }).start();
  };

  const hidePassword = (): void => {
    setState({...state, hide: !state.hide});
  };

  const animate = {
      fontSize: state.animated.interpolate({
        inputRange: Platform.OS === 'ios' ? [-1, 2] : [0, 1],
        outputRange: Platform.OS === 'ios' ? [-2, 1] : [0, 1],
      }),
      fontFamily: 'Lato-Semibold',
      marginBottom: 5,
      color: colors.gray4,
    },
    styleTextInput = {
      width: isPassword ? '90%' : '100%',
      borderRadius: sizes.radiusMedium,
      fontFamily: 'Lato-Regular',
      backgroundColor: colors.backgroundColor,
      padding: 10,
      borderColor: state.value !== '' ? colors.success : colors.gray2,
      fontSize: fs(7, width),
    },
    containerText: Object = {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: colors.backgroundColor,
      borderRadius: 5,
      height: heightPercentageToDP(6.5),
    },
    container = [containerStyle];

  return (
    <View style={container}>
      <Animated.Text style={animate}>{`${placeholder}`}</Animated.Text>
      <View style={containerText}>
        <TextInput
          multiline={multiline}
          keyboardType={isNumber ? 'phone-pad' : 'default'}
          placeholder={`${placeholder}`}
          autoCapitalize={uppercase ? 'characters' : 'none'}
          value={state.value}
          onChangeText={(value: any) => handleType(value)}
          style={[styleTextInput, style]}
          secureTextEntry={isPassword ? state.hide : false}
        />
        {isPassword && (
          <Button onPress={hidePassword}>
            <Icon
              name={state.hide ? 'eye-off' : 'eye'}
              color="grey"
              size={fs(13, width)}
            />
          </Button>
        )}
      </View>
    </View>
  );
};

CustomTextInput.defaultProps = {
  placeholder: 'Input here',
  multiline: false,
  uppercase: false,
};

export default CustomTextInput;
