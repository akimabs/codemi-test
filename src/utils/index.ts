import {colors, images} from '../theme';

const startedScreen = [
  {
    imageUri: images.Onboarding.image1,
    heading: 'Pembelajaran Online',
    description: 'Akses module pelatihan kapan sana dan di mana saja.',
    key: 'first',
    color: colors.primary2 + 60,
  },
  {
    imageUri: images.Onboarding.image2,
    heading: 'Pembelajaran Kolaboratif',
    description:
      'Dapatkan pengalaman belajar secara kolaboratif dengan karyawan lainnya',
    key: 'second',
    color: colors.primary2 + 60,
  },
  {
    imageUri: images.Onboarding.image3,
    heading: 'Pembelajaran di Dalam Kelas',
    description:
      'Pelatihan bisa dilakukan di dalam kelas konvensional yang terdiri dari instruktur dan peserta',
    key: 'third',
    color: colors.primary + 60,
  },
];

export {startedScreen};
