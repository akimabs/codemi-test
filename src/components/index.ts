import Button from './Button';
import Text from './Text';
import TextInput from './TextInput';
import CustomView from './View';
import {showErrorToast, showInfoToast, showSuccessToast} from './Toast';

export {
  Button,
  Text,
  TextInput,
  CustomView,
  showErrorToast,
  showInfoToast,
  showSuccessToast,
};
