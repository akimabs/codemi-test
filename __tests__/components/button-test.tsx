/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import {Button, Text} from '../../src/components/';

describe('Button', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Button', () => {
      const button = renderer.create(
          <Button type="primary" title="Button Primary" />,
        ),
        butttonLight = renderer.create(
          <Button type="light" title="Button Light" />,
        ),
        buttonSuccess = renderer.create(
          <Button type="success" title="Button Success" />,
        ),
        buttenDanger = renderer.create(
          <Button type="danger" title="Button Danger" />,
        ),
        withOutRipple = renderer.create(
          <Button
            type="primary"
            title="Button Primary"
            withOutRipple={false}
            withOutAnimate
          />,
        ),
        withOutAnimate = renderer.create(
          <Button
            type="primary"
            title="Button Primary"
            withOutAnimate
            withOutRipple={false}
          />,
        ),
        buttonWithChildern = renderer.create(
          <Button>
            <Text type="thin">Button Primary</Text>
          </Button>,
        ),
        disable = renderer.create(
          <Button
            disable
            withOutAnimate
            withOutRipple
            type="primary"
            title="Disable Button Primary"
          />,
        ),
        disableButtonWithChild = renderer.create(
          <Button disable withOutAnimate withOutRipple type="primary">
            <Text type="thin">Button Primary</Text>
          </Button>,
        );

      expect({
        button,
        butttonLight,
        buttonSuccess,
        buttenDanger,
        withOutRipple,
        withOutAnimate,
        buttonWithChildern,
        disable,
        disableButtonWithChild,
      }).toMatchSnapshot('button snapshot');
    });
  });
});
