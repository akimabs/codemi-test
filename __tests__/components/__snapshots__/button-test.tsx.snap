// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot - Button: button snapshot 1`] = `
Object {
  "buttenDanger": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d9534f",
            "borderRadius": 8,
            "height": 80,
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                Object {
                  "color": "#FFF",
                  "fontFamily": "Ubuntu-Medium",
                },
                false,
                false,
                false,
              ],
            ]
          }
          type="bold"
        >
          Button Danger
        </Text>
      </View>
    </View>
  </View>,
  "button": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#32ab3a",
            "borderRadius": 8,
            "height": 80,
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                false,
                Object {
                  "color": "#FFF",
                  "fontFamily": "Ubuntu-Medium",
                },
                false,
                false,
              ],
            ]
          }
          type="bold"
        >
          Button Primary
        </Text>
      </View>
    </View>
  </View>,
  "buttonSuccess": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5cb85c",
            "borderRadius": 8,
            "height": 80,
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "color": "#FFF",
                  "fontFamily": "Ubuntu-Medium",
                },
              ],
            ]
          }
          type="bold"
        >
          Button Success
        </Text>
      </View>
    </View>
  </View>,
  "buttonWithChildern": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={Object {}}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                Object {
                  "fontFamily": "Lato-Thin",
                  "fontSize": 16,
                },
                false,
                false,
                false,
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              undefined,
            ]
          }
          type="thin"
        >
          Button Primary
        </Text>
      </View>
    </View>
  </View>,
  "butttonLight": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#32ab3a30",
            "borderRadius": 8,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                false,
                false,
                Object {
                  "color": "#32ab3a",
                  "fontFamily": "Ubuntu-Medium",
                },
                false,
              ],
            ]
          }
          type="bold"
        >
          Button Light
        </Text>
      </View>
    </View>
  </View>,
  "disable": <View>
    <View
      style={
        Array [
          Array [
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "#32ab3a",
              "borderRadius": 8,
              "height": 80,
              "justifyContent": "center",
              "overflow": "hidden",
            },
            false,
            false,
          ],
          undefined,
          Object {
            "transform": Array [
              Object {
                "scale": Object {
                  "current": 1,
                },
              },
            ],
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#072c5a",
              "fontSize": 12,
            },
            Array [
              false,
              false,
              false,
              Object {
                "fontFamily": "Lato-Bold",
                "fontSize": 16,
              },
            ],
            Object {
              "color": "#333333",
            },
            Object {
              "fontSize": 14,
            },
            Array [
              false,
              Object {
                "color": "#FFF",
                "fontFamily": "Ubuntu-Medium",
              },
              false,
              false,
            ],
          ]
        }
        type="bold"
      >
        Disable Button Primary
      </Text>
    </View>
  </View>,
  "disableButtonWithChild": <View>
    <View
      style={
        Array [
          Array [
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "#32ab3a",
              "borderRadius": 8,
              "height": 80,
              "justifyContent": "center",
              "overflow": "hidden",
            },
            false,
            false,
          ],
          undefined,
          Object {
            "transform": Array [
              Object {
                "scale": Object {
                  "current": 1,
                },
              },
            ],
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#072c5a",
              "fontSize": 12,
            },
            Array [
              Object {
                "fontFamily": "Lato-Thin",
                "fontSize": 16,
              },
              false,
              false,
              false,
            ],
            Object {
              "color": "#333333",
            },
            Object {
              "fontSize": 14,
            },
            undefined,
          ]
        }
        type="thin"
      >
        Button Primary
      </Text>
    </View>
  </View>,
  "withOutAnimate": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Array [
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "#32ab3a",
                "borderRadius": 8,
                "height": 80,
                "justifyContent": "center",
                "overflow": "hidden",
              },
              false,
              false,
            ],
            undefined,
            Object {
              "transform": Array [
                Object {
                  "scale": Object {
                    "current": 1,
                  },
                },
              ],
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                false,
                Object {
                  "color": "#FFF",
                  "fontFamily": "Ubuntu-Medium",
                },
                false,
                false,
              ],
            ]
          }
          type="bold"
        >
          Button Primary
        </Text>
      </View>
    </View>
  </View>,
  "withOutRipple": <View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Array [
              false,
              Object {
                "alignItems": "center",
                "backgroundColor": "#32ab3a",
                "borderRadius": 8,
                "height": 80,
                "justifyContent": "center",
                "overflow": "hidden",
              },
              false,
              false,
            ],
            undefined,
            Object {
              "transform": Array [
                Object {
                  "scale": Object {
                    "current": 1,
                  },
                },
              ],
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#072c5a",
                "fontSize": 12,
              },
              Array [
                false,
                false,
                false,
                Object {
                  "fontFamily": "Lato-Bold",
                  "fontSize": 16,
                },
              ],
              Object {
                "color": "#333333",
              },
              Object {
                "fontSize": 14,
              },
              Array [
                false,
                Object {
                  "color": "#FFF",
                  "fontFamily": "Ubuntu-Medium",
                },
                false,
                false,
              ],
            ]
          }
          type="bold"
        >
          Button Primary
        </Text>
      </View>
    </View>
  </View>,
}
`;
